.github/workflows/test.yml

```diff
@@ -0,0 +1,94 @@
+# TFS-API Git Hub Actions
+#####################################
+# This GHA is responsible for initiating various test suites
+# upon a PR creation. Following test suites are executed:
+# 1. Unit tests (Jest)
+# 2. API tests (Jest)
+
+name: Quality Assurance - Tests
+run-name: Executing test QA on ${{ github.repository }} 🚀
+
+on:
+  pull_request:
+    branches: [ main ]
+    paths:
+    - '**'
+
+env:
+  environment: dev
+  timezone: 'Europe/London'
+  node: '18'
+
+jobs:
+
+# 1. Setup test infrastructure
+  setup:
+    name: Infrastructure setup 🔧
+    runs-on: ubuntu-latest
+    outputs:
+      environment: ${{ env.environment }}
+      timezone: ${{ env.timezone }}
+    steps:
+    - name: Environment 🧪
+      run: echo "Environment set to ${{ env.environment }}"
+  
+    - name: Timezone 🌐
+      run: echo "Timezone set to ${{ env.timezone }}"
+
+# 2. Unit tests
+  unit-tests:
+    name: Unit 👷
+    needs: setup
+    environment:
+      name: ${{ needs.setup.outputs.environment }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Timezone
+        uses: szenius/set-timezone@v1.0
+        with:
+          timezoneLinux: ${{ needs.setup.outputs.timezone }}
+
+      - name: Repository
+        uses: actions/checkout@v3
+
+      - name: Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ env.node }}
+
+      - name: Dependencies
+        working-directory: ./
+        run: npm ci
+
+      - name: Execute
+        working-directory: ./
+        run: npm run unit-test
+
+# 3. API tests
```

Comment:
I'm not sure we are able to run this without hammering their servers?

Reply:
No unit-test currently at the moment, are we expecting to use `unit-test` for helper functions?

Reply:
unit test should be fine. but isn't this API tests?

Reply:
Ah! okay will need to re-visit this when more tests are written. Shall I leave it in for now and then we can measure when more tests are written?

Reply:
sounds good